site/content/api/Command-And-Control-API.md

```diff
@@ -3,38 +3,129 @@ title = "Command & Control API"
 
 There is also a description of Command & Control [at the conceptual level]({{< relref "/concepts/command-and-control.md" >}}) 
 and the [Getting Started]({{< ref "getting-started" >}}) guide contains a walk through example.
 
+# Preconditions
+## Preconditions for sending a command or notification command
+
+1. The *Business Application* has established an AMQP connection with the AMQP 1.0 Network.  
+2. The *Business Application* has established an AMQP link in role *sender* with the target address `control/${tenant_id}/${device_id}`, where `${device_id}` is the ID of the device to send messages to and `${tenant_id}` is the ID of the tenant that the device belongs to.
+
+**Link establishment**
+
+The following sequence diagram shows the necessary links that need to be established:
+
+![Send Command Preconditions](../command_control_send_preconditions.png)
+
+## Preconditions for receiving a command response
+
+*Command* messages (in contrast to *notification command* messages) need to be responded by the device. For this purpose,
+the *Business Application* needs to open a receiver link as precondition: 
+
+1. The *Business Application* has established an AMQP link in role *receiver* with the source address `control/${tenant_id}/${reply-id}`.
+This link is used by the *Business Application* to receive the response of the executed command. This link’s source address is also referred to as `the reply-to address` for the request messages. 
+2. The `${reply-id}` may be any arbitrary string chosen by the client.  
+
+To scope a response to the device the `${reply-id}` should consist of the device-id plus an arbitrary string to scope the command sender instance. If all responses shall be received just in the context of the tenant, the device-id has to be omitted.   
+
+**Link establishment**
+
+The following sequence diagram shows the necessary links that need to be established to enable sending back a response to the Business Application:
+
+![Send Command](../command_control_receive_preconditions.png)
+
+# Outcomes for sending a command or notification command
+Commands and notification commands that are forwarded by the AMQP 1.0 network to a protocol adapter need to be responded with following outcomes:
+
+| Processing in protocol adapter   | AMQP outcome                  |
+|----------------------------------|------------------------------ |
+| Bad command                      | Rejected ("hono:bad-request") |
+| Device not connected (anymore)   | Release                       | 
+| Command delivered                | Accepted                      | 
+
+For any outcome except for `Accepted` the *Business application* should immediately close the response receiver link again, 
+if it was scoped to exactly one command response. If the receiver link is scoped to the tenant, it usually should remain 
+open.
+
```

Comment:
Why should the application close the receiver link for e.g. device 4711 if a release happened or the command was bad? It could be a device, which is connected by MQTT, but was unconnected for a short time or one command was wrong. In both cases it would be fine to stay with the link.

I think you have the HTTP adapter case in mind were only one command could be send per ttd event from the device. 

It seems that we cannot distinguish this case on the side of the application at the moment from the variant that basically any number of commands can be sent after one ttd.